File: Frontend/src/interfaces/response-dtos/user-dto.ts

import { FolderDto } from "./folder-dto"
import { AudioDto } from "../creation-dtos/audio-dto"
import { EbookDto } from "../creation-dtos/ebook-dto"
import { ImageDto } from "../creation-dtos/image-dto"
import { VideoDto } from "../creation-dtos/video-dto"

export interface UserDto {
    id?: number,
    name?: string,
    email?: string,
    registrationDate?: Date
    folders?: FolderDto[]
    audios?: AudioDto[]
    videos?: VideoDto[]
    images?: ImageDto[]
    ebooks?: EbookDto[]
}

File: Backend/Media_library/Dtos/ResponseDtos/UserDto.cs
using Media_library.Entities;
using Media_library.Entities.MediaTypes;

namespace Media_library.Dtos.ResponseDtos;

public class UserDto
{
    public Guid Id { get; set; }
    public string Name { get; set; }
    public string Email { get; set; }
    public DateTime RegistrationDate { get; set; }
    public IEnumerable<FolderDto> Folders { get; set; }
    public IEnumerable<AudioDto> Audios { get; set; }
    public IEnumerable<VideoDto> Videos { get; set; }
    public IEnumerable<ImageDto> Images { get; set; }
    public IEnumerable<EbookDto> Ebooks { get; set; }
}